### PR TITLE
sd-login: add two interfaces for retriving session info

### LINKED ISSUE #26539: Provide login name and timestamp via sd-login.h API to replace utmp
### Component

systemd-logind

### Is your feature request related to a problem? Please describe

`/run/tmp` uses even on many 64bit systems int32_t as time_t. This is for compatibility with 32bit userland. Which leads to the problem, that utmp/wtmp/btmp and lastlog will not survive Y2038 in the current form. The current utmp implementation has additional problems, too. That's why the glibc developers made pretty clear that they will not "fix" the utmp interface but deprecate it.

systemd-logind has all information and provides them via different interfaces. A PoC shows that this could replace the utmp interface. But the sd-login.h API misses two functions: to get the name of the user and the login timestamp. This are the most common used information from the utmp file.

Some background and consumers of the utmp interface can be found in the following document, which is WiP and far from complete:
https://github.com/thkukuk/utmpx/blob/main/Y2038.md

### Describe the solution you'd like

Most important: provide a function which returns the login time of the user. That's used by nearly every application using utmp data.

An additional nice to have interface would be.
```
/* Determine user name of session */
int sd_session_get_user_name(const char *session, char *name);
```
This function is like sd_session_get_uid, but returns the name and not the UID. 

### Describe alternatives you've considered

For the  sd_session_get_user_name() function, sd_session_get_uid() could be used, which would require that we do everytime a getpwuid() lookup for something which logind has.

Else, for both functions, a new library using the DBUS interface could be written, but that would mean a higher efford for something, where for the majority we have already existing functions.

### The systemd version you checked that didn't have the feature you are asking for

Current git.

### PR DESCRIPTION
**Pull request by YHNdnzj**
As per requested in https://github.com/systemd/systemd/issues/26539

Closes #26539

### CODE REVIEW

File: man/sd_session_is_active.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <refname>sd_session_get_uid</refname>
     <refname>sd_session_get_username</refname>
     <refname>sd_session_get_seat</refname>
+    <refname>sd_session_get_open_time</refname>
```
Comment (poettering) [inline man/sd_session_is_active.xml]: hmm, is "open time" the right name for this? maybe "start_time"?

File: src/libsystemd/sd-login/sd-login.c [hunk 2]
```diff
@@ -750,6 +750,33 @@ _public_ int sd_session_get_seat(const char *session, char **seat) {
         return session_get_string(session, "SEAT", seat);
 }
 
+_public_ int sd_session_get_start_time(const char *session, uint64_t *usec) {
```
Comment (poettering) [inline src/libsystemd/sd-login/sd-login.c:753]: the two new symbols must be added src/libsystemd/libsystemd.sym, too

**Comment by thkukuk** [comment]
Thanks, looks very similar to my patch which I'm currently testing, didn't expect that somebody else jumps so fast on the train. So looks good for me.
I found out that a third function would be helpful: sd_uid_get_login_time(uid_t uid, uint64_t *usec), as this timestamp could be older than the one from the sessions, if the oldest session of this user got closed meanwhile. But I can make a separate PR for this if this one gets merged.

**Review by yuwata — APPROVED** [review]

**Pull request merged**